From 0b829a8cf68f7ed27b88bd20c1458b53641609af Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:41 -0400
Subject: [block] blk-mq: ->timeout should be cleared in blk_mq_rq_ctx_init()

Message-id: <1402670238-13916-114-git-send-email-snitzer@redhat.com>
Patchwork-id: 83894
O-Subject: [RHEL7.1 PATCH 113/150] blk-mq: ->timeout should be cleared in blk_mq_rq_ctx_init()
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit f6be4fb4bcb396fc3b1c134b7863351972de081f
Author: Jens Axboe <axboe@fb.com>
Date:   Fri Jun 6 11:03:48 2014 -0600

    blk-mq: ->timeout should be cleared in blk_mq_rq_ctx_init()

    It'll be used in blk_mq_start_request() to set a potential timeout
    for the request, so clear it to zero at alloc time to ensure that
    we know if someone has set it or not.

    Fixes random early timeouts on NVMe testing.

    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 75fc33f..fafea52 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -204,6 +204,8 @@ static void blk_mq_rq_ctx_init(struct request_queue *q, struct blk_mq_ctx *ctx,
  rq->sense = NULL;
 
  INIT_LIST_HEAD(&rq->timeout_list);
+ rq->timeout = 0;
+
  rq->end_io = NULL;
  rq->end_io_data = NULL;
  rq->next_rq = NULL;
-- 
1.7.1